[4.3.0] docs(filemanager): SelectedItems documentation (#1483)
telerik/blazor-docs: components/filemanager/selection.md

---
title: Selection
page_title: FileManager Selection
description: Item selection in the FileManager for Blazor.
slug: filemanager-selection
tags: telerik,blazor,filemanager,selection,item,file,folder,selected,items
published: True
position: 10
---

# FileManager Selection

The FileManager lets the user select one or multiple files and folders in its main pane. You can also pre-select a desired item.

> To get or set the selected items, use the `SelectedItems` parameter. It is a collection of items from the FileManager's Data.

Reviewer: ```suggestion
To get or set the selected items, use the `SelectedItems` parameter. It is a collection of items from the FileManager's `Data`.
```

> The `SelectedItems` parameter allows two-way binding (`@bind-SelectedItems`) and [one-way binding] + [SelectedItemsChanged]({%slug filemanager-events%}#selecteditemschanged) event.

Reviewer: ```suggestion
The `SelectedItems` parameter supports two-way binding (`@bind-SelectedItems`) and one-way binding together with the [`SelectedItemsChanged`]({%slug filemanager-events%}#selecteditemschanged) event.
```

                DateModified = new DateTime(2022, 2, 13),
                DateModifiedUtc = new DateTime(2022, 2, 13),
                Path = Path.Combine(Images.Path, "dashboard-design.png"),
                Size = 1024
            };

        var gridDesign = new FlatFileEntry()
            {
                Id = "7",
                ParentId = Images.Id,
                Name = "Grid Design",
                IsDirectory = false,
                HasDirectories = false,
                Extension = ".png",
                DateCreated = new DateTime(2022, 1, 12),
                DateCreatedUtc = new DateTime(2022, 1, 12),
                DateModified = new DateTime(2022, 2, 13),
                DateModifiedUtc = new DateTime(2022, 2, 13),
                Path = Path.Combine(Images.Path, "grid-design.jpg"),
                Size = 1024
            };

        var files = new List<FlatFileEntry>()
            {
                workFiles,

                Documents,
                specification,
                report,

                Images,
                dashboardDesign,
                gridDesign
            };

        return await Task.FromResult(files);
    }
}
````

> ## One-way binding of the SelectedItems

Reviewer 2: ```suggestion
## One-Way Binding of the SelectedItems
```
